src/Components/WebAssembly/Server/src/RazorComponentsBuilderExtensions.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Endpoints;
+using Microsoft.AspNetCore.Components.Web;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+/// <summary>
+/// Extension methods to configure an <see cref="IServiceCollection"/> for WebAssembly components.
+/// </summary>
+public static class RazorComponentsBuilderExtensions
+{
+    /// <summary>
+    /// Adds services to support rendering interactive WebAssembly components in a razor components
+    /// application.
+    /// </summary>
+    /// <param name="builder">The <see cref="IRazorComponentsBuilder"/>.</param>
+    /// <param name="configure">A callback to configure <see cref="WebAssemblyComponentsEndpointOptions"/>.</param>
+    /// <returns>An <see cref="IRazorComponentsBuilder"/> that can be used to further customize the configuration.</returns>
+    public static IRazorComponentsBuilder AddWebAssemblyComponents(this IRazorComponentsBuilder builder, Action<WebAssemblyComponentsEndpointOptions>? configure = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder, nameof(builder));
+
+        builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<RenderModeEndpointProvider, WebAssemblyEndpointProvider>());
+
+        if (configure is not null)
+        {
+            builder.Services.Configure(configure);
+        }
+
+        return builder;
+    }
+
+    private class WebAssemblyEndpointProvider : RenderModeEndpointProvider
```

Comment:
We are missing two extension methods `'AddWebAssemblyRenderMode', 'AddServerRenderMode'` to plug a render mode independent of discovery.

Reply:
Do you think it's fine if I add those as part of this PR?

Reply:
Looks like those are [already a thing](https://github.com/dotnet/aspnetcore/blob/3f389bb07e872cd11b7b7b401ec673465bf89600/src/Components/Endpoints/src/Builder/RazorComponentEndpointConventionBuilder.cs#L47), actually.